pkg/volume/local/local.go

```diff
@@ -307,3 +365,44 @@ func (u *localVolumeUnmounter) TearDownAt(dir string) error {
 	glog.V(4).Infof("Unmounting volume %q at path %q\n", u.volName, dir)
 	return util.UnmountMountPoint(dir, u.mounter, true) /* extensiveMountPointCheck = true */
 }
+
+// localVolumeMapper implements the BlockVolumeMapper interface for local volumes.
+type localVolumeMapper struct {
+	*localVolume
+	readOnly bool
+}
+
+var _ volume.BlockVolumeMapper = &localVolumeMapper{}
+
+// SetUpDevice provides physical device path for the local PV.
+func (m *localVolumeMapper) SetUpDevice() (string, error) {
+	return filepath.EvalSymlinks(m.globalPath)
```

Comment:
Why do we need to eval symlinks?

Reply:
@msau42  The primary consumer of this function seems to be this line: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go#L855

The returned path is used as the source for creating symbolic links at lines 874 and 887. I thought it would be better that they got created using the true device as source, rather than symbolic link to symbolic link.  Let me know if this is not ideal.

Reply:
My only concern is would it make debugging slightly harder?  Because if you try searching for the path in the spec, you won't see it if it got resolved to something else.

Reply:
Fixed.